src/Http/Http/src/TagsAttribute.cs

```diff
@@ -0,0 +1,28 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+namespace Microsoft.AspNetCore.Http
+{
+    /// <summary>
+    /// Specifies a collection of tags in <see cref="Endpoint.Metadata"/>.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Method | AttributeTargets.Delegate | AttributeTargets.Class, Inherited = false, AllowMultiple = false)]
+    public sealed class TagsAttribute : Attribute, ITagsMetadata
+    {
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="tags"></param>
+        public TagsAttribute(params string[] tags)
+        {
+            Tags = new List<string>(tags);
+        }
+
+        /// <summary>
+        /// Gets the collection of tags associated with the endpoint.
+        /// </summary>
+        public IReadOnlyList<string> Tags { get; }
+    }
```

Comment:
Helps https://github.com/dotnet/aspnetcore/issues/34604 in order to discover the endpoint metadata.
```suggestion
    }
    
    /// <inheritdoc />
    public override string ToString()
    {
        return $"Tags: {string.Join(",", Tags)}";
    }
```

Reply:
Thanks for the recommendation! I think it'll be easier if we batch all the changes related to #34604 above into the same PR and review collectively.

Also, I suspect that maybe we want to use a `DebuggerDisplay` string for this as well.